src/main/java/com/es/phoneshop/model/product/ArrayListProductDao.java

```diff
@@ -1,25 +1,84 @@
+        products.add(new Product(6L, "htces4g", "HTC EVO Shift 4G", new BigDecimal(320), usd, 3,
+                "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/HTC/HTC%20EVO%20Shift%204G.jpg"));
+        products.add(new Product(7L, "sec901", "Sony Ericsson C901", new BigDecimal(420), usd, 30,
+                "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/Sony/Sony%20Ericsson%20C901.jpg"));
+        products.add(new Product(8L, "xperiaxz", "Sony Xperia XZ", new BigDecimal(120), usd, 100,
+                "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/Sony/Sony%20Xperia%20XZ.jpg"));
+        products.add(new Product(9L, "nokia3310", "Nokia 3310", new BigDecimal(70), usd, 100,
+                "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/Nokia/Nokia%203310.jpg"));
+        products.add(new Product(10L, "palmp", "Palm Pixi", new BigDecimal(170), usd, 30,
+                "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/Palm/Palm%20Pixi.jpg"));
+        products.add(new Product(11L, "simc56", "Siemens C56", new BigDecimal(70), usd, 20,
+                "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/Siemens/Siemens%20C56.jpg"));
+        products.add(new Product(12L, "simc61", "Siemens C61", new BigDecimal(80), usd, 30,
+                "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/Siemens/Siemens%20C61.jpg"));
+        products.add(new Product(13L, "simsxg75", "Siemens SXG75", new BigDecimal(150), usd, 40,
+                "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/Siemens/Siemens%20SXG75.jpg"));
+    }
 
-public class ArrayListProductDao implements ProductDao {
     @Override
-    public Product getProduct(Long id) {
-        throw new RuntimeException("Not implemented");
+    public Product getProduct(Long id) throws ProductNotFoundException
+    {
+        return products.stream()
+                .filter(p -> p.getId().equals(id))
+                .findFirst()
+                .orElseThrow(() -> new ProductNotFoundException(id));
     }
 
     @Override
-    public List<Product> findProducts() {
-        throw new RuntimeException("Not implemented");
+    public List<Product> findProducts()
+    {
+        return products.stream()
+                .filter(product -> product.getPrice() != null)
+                .filter(product -> product.getStock() > 0)
+                .collect(Collectors.toList());
     }
 
     @Override
-    public void save(Product product) {
-        throw new RuntimeException("Not implemented");
+    public synchronized void save(Product product)
+    {
+        long[] ids = products.stream().mapToLong(Product::getId).toArray();
```

Comment:
Red rid of IdSeeker class as it doesn't look fitting into web project architecture. The idea of dao generating ids itself is good though, you can just use static variable and increment it in here (and also add sample products via save() method in the beginning). Note that in this case you would need to remove id from Product constructor signature (as it will be assigned by dao)